sui_core/src/authority_client.rs

```diff
@@ -68,6 +73,9 @@ pub trait AuthorityAPI {
         &self,
         request: CheckpointRequest,
     ) -> Result<CheckpointResponse, SuiError>;
+
+    #[cfg(test)]
+    fn as_any_mut(&mut self) -> &mut dyn Any;
```

Comment:
This does not work, as `#[cfg(test]` is isolated to only exist for unit tests hanging off of this crate. Also, optionally adding functions to traits is generally frowned upon.

Reply:
this exists to satisfy exactly one test which was effectively downcasting SafeClient, and that test still works. this doesn't have to work beyond that.